More smart pointer adoption in WebViewImpl.mm

rwlbuis opened the pull request:
#### 3557b5441299e9195c2172aa57da49df43e86e4c
<pre>
More smart pointer adoption in WebViewImpl.mm
<a href="https://bugs.webkit.org/show_bug.cgi?id=296556">https://bugs.webkit.org/show_bug.cgi?id=296556</a>

Reviewed by Chris Dumez.

Apply <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines</a> based on
clang-tidy warnings.

* Source/WebCore/PAL/pal/spi/mac/NSWindowSPI.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(-[WKMouseTrackingObserver mouseMoved:]):
(-[WKMouseTrackingObserver mouseEntered:]):
(-[WKMouseTrackingObserver mouseExited:]):
(-[WKResponderChainSink detach]):
(-[WKTextListTouchBarViewController _selectList:]):
(-[WKTextTouchBarItemController candidateListTouchBarItem:endSelectingCandidateAtIndex:]):
(-[WKTextTouchBarItemController candidateListTouchBarItem:changedCandidateListVisibility:]):
(-[WKTextTouchBarItemController touchBarDidExitCustomization:]):
(-[WKTextTouchBarItemController didChangeAutomaticTextCompletion:]):
(-[WKTextTouchBarItemController _wkChangeTextAlignment:]):

Canonical link: <a href="https://commits.webkit.org/299322@main">https://commits.webkit.org/299322@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2d68e439b9a4ea2a2c3a9c5b332af7040e03ba2d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118594 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38275 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28926 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124771 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70653 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120472 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38971 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46857 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90009 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59555 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121547 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31037 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106320 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70513 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30092 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24431 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68428 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100475 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24622 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127832 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45501 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34329 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98641 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45865 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102541 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98425 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25029 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43867 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21858 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41999 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45371 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/51049 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44834 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48181 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46521 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a76024fc07abdec8b23019cf616fbd124e6814c7. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48597

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/fa6b8171e075b6593a53605bf01bcbb97045a317. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48597

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/709874bda6f7c5119887ec70365f03aac18a591e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48597

cdumez commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -637,7 +637,7 @@ - (BOOL)isFlipped
 
 @interface WKResponderChainSink : NSResponder {
-    NSResponder *_lastResponderInChain;
+    RetainPtr<NSResponder> _lastResponderInChain;
```
I am a little worried about reference cycles here, maybe we can use a WeakObjCPtr here?

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2d68e439b9a4ea2a2c3a9c5b332af7040e03ba2d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48597

rwlbuis committed:
More smart pointer adoption in WebViewImpl.mm
https://bugs.webkit.org/show_bug.cgi?id=296556

Reviewed by Chris Dumez.

Apply https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines based on
clang-tidy warnings.

* Source/WebCore/PAL/pal/spi/mac/NSWindowSPI.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(-[WKMouseTrackingObserver mouseMoved:]):
(-[WKMouseTrackingObserver mouseEntered:]):
(-[WKMouseTrackingObserver mouseExited:]):
(-[WKResponderChainSink detach]):
(-[WKTextListTouchBarViewController _selectList:]):
(-[WKTextTouchBarItemController candidateListTouchBarItem:endSelectingCandidateAtIndex:]):
(-[WKTextTouchBarItemController candidateListTouchBarItem:changedCandidateListVisibility:]):
(-[WKTextTouchBarItemController touchBarDidExitCustomization:]):
(-[WKTextTouchBarItemController didChangeAutomaticTextCompletion:]):
(-[WKTextTouchBarItemController _wkChangeTextAlignment:]):

Canonical link: https://commits.webkit.org/299322@main
